values.go

```diff
@@ -37,6 +37,12 @@ const (
 	ModeHalf              = "half"
 )
 
+// deferredSet is used during marshaling to properly populate reused non-pointer struct references
+type deferredSet struct {
+	dest interface{}
+	sets []func(interface{})
```

Comment:
i decided on anonymous functions at the start because number of things to close on plus number of operations done was tbd. makes sense since the result was one operation on one closed item should ditch funcs.